src/epcpm/sunspectocsv.py

```diff
@@ -0,0 +1,532 @@
+                model_offset=self.model_offset,
+                is_table=self.is_table,
+                repeating_block_reference=self.repeating_block_reference,
+                address_offset=self.address_offset + summed_increments,
+            )
+            built_rows, address_offset_increment = builder.gen()
+            summed_increments += address_offset_increment
+            rows.append(built_rows)
+
+        return rows, summed_increments
+
+
+@builders(epcpm.sunspecmodel.DataPointBitfield)
+@attr.s
+class DataPointBitfield:
+    wrapped = attr.ib()
+    model_type = attr.ib()
+    scale_factor_from_uuid = attr.ib()
+    parameter_uuid_finder = attr.ib()
+    model_id = attr.ib()
+    model_offset = attr.ib()
+    is_table = attr.ib()
+    repeating_block_reference = attr.ib()
+    address_offset = attr.ib()
+
+    def gen(self):
+        row = Fields()
+        row.address_offset = self.address_offset
+        row.modbus_address = self.model_offset + self.address_offset
+        row.field_type = self.model_type
+
+        for name, field in attr.asdict(bitfield_fields).items():
+            if field is None:
+                continue
+
+            setattr(row, name, getattr(self.wrapped, field.name))
+
+        if row.type is not None:
+            row.type = self.parameter_uuid_finder(row.type).name
+
+        if self.wrapped.parameter_uuid is not None:
+            parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
+
+            row.label = parameter.name
+            row.name = parameter.abbreviation
+            if row.units is None:
+                row.units = parameter.units
+            row.description = parameter.comment
+            row.read_write = "R" if parameter.read_only else "RW"
+            row.parameter_uuid = parameter.uuid
+            row.parameter_uses_interface_item = False
+            row.type_uuid = self.wrapped.type_uuid
+            row.not_implemented = False
+            row.uuid = self.wrapped.uuid
+            row.class_name = "DataPointBitfield"
+
+        uses_interface_item = (
+            isinstance(parameter, epyqlib.pm.parametermodel.Parameter)
+            and parameter.uses_interface_item()
+        )
+
+        return row, row.size
+
+
+@builders(epcpm.sunspecmodel.TableRepeatingBlockReference)
+@attr.s
+class TableRepeatingBlockReference:
+    wrapped = attr.ib()
+    add_padding = attr.ib()
+    padding_type = attr.ib()
+    model_type = attr.ib()
+    model_id = attr.ib()
+    model_offset = attr.ib()
+    address_offset = attr.ib()
+    parameter_uuid_finder = attr.ib(default=None)
+
+    def gen(self):
+        table_repeating_block = self.wrapped.original
+
+        rows = []
+        block_size = 0
+        for curve_index in range(table_repeating_block.repeats):
+            curve_points = 0
+            for point in table_repeating_block.children:
+                table_element1 = self.parameter_uuid_finder(point.parameter_uuid)
```

Comment:
provide some description on whats going on

Reply:
Comment added in 004cc5dfe8e5765d8f063afd11dec72e1e9f2840. If that is not enough information, context is given in the SunSpec and Static Modbus document.